app/src/main/kotlin/games/Board.kt

```diff
@@ -41,22 +41,42 @@ class Board :ValidMoveProvider{
         elements[move.row][move.column].setState(player.id)
     }
 
+    internal fun setState(gameState: String) {
+        val it = gameState.splitToSequence(" ", "\n").iterator()
+        for (r in 0..2) {
+            for (c in 0..2) {
+                elements[r][c].setState(toState(it.next()))
+            }
+        }
+    }
```

Comment:
This new method is just used in testing, which is why the function is marked as `internal`. I don't know if this is best practice. I have a feeling it's not.